encoder.go

```diff
@@ -2,9 +2,9 @@ package midi
 
 import (
 	"encoding/binary"
-	"errors"
 	"io"
-	"log"
+
+	"github.com/pkg/errors"
```

Comment:
adding this dependency what was preventing me from merging this PR (that I honestly forgot about). Any chance we can avoid adding an external dependency?